Android - NPE on cancel purchase

Hello!

I have found an NPE here are the logs:

> Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.smartmobilesoftware.util.Purchase.getSku()' on a null object reference
> at jp.wizcorp.wizpurchase.WizPurchase$10.onIabPurchaseFinished(WizPurchase.java:727)
> 

It happen when I start a purchase, and cancel it buy clicking outside de Google Play window.

File:
https://github.com/Wizcorp/cordova-plugin-wizpurchase/blob/master/src/android/WizPurchase.java#L727

I thought about make a PR with a
`if(purchase != null)`

But not sure if it's normal that purchase is null here...



Regards,

--- a/src/android/WizPurchase.java
+++ b/src/android/WizPurchase.java
@@ -724,41 +724,43 @@ public class WizPurchase extends CordovaPlugin {
 		public void onIabPurchaseFinished(IabResult result, Purchase purchase) {
 			Log.d(TAG, "Purchase finished: " + result + ", purchase: " + purchase);
 
-			setStartedPurchase(purchase.getSku(), false);
-
-			// Check if we have the handler
-			if (mMakePurchaseCbContext != null) {
-				// Check if there are purchase errors
-				if (result.isFailure()) {
+			// Check if there are purchase errors
+			if (result.isFailure()) {
+				// Check if we have the handler
+				if (mMakePurchaseCbContext != null) {
 					// Get the Error message
 					int errorCode = returnErrorCode(result.getResponse());
 					// Dispatch the Error event
 					mMakePurchaseCbContext.error(errorCode);
 					// Clear the handler and stop processing
 					mMakePurchaseCbContext = null;
-					return;
 				}
-
+			} else {
 				Log.d(TAG, "Purchase successful.");
 
-				try {
-					// Build the return Object
-					JSONObject purchaseResult = convertToJSONObject(purchase);
-					// Return the object
-					mMakePurchaseCbContext.success(purchaseResult);
-				} catch (JSONException e) { }
-				// Clear the handler
-				mMakePurchaseCbContext = null;
+				setStartedPurchase(purchase.getSku(), false);
+
+				// Check if we have the handler
+				if (mMakePurchaseCbContext != null) {
+					try {
+						// Build the return Object
+						JSONObject purchaseResult = convertToJSONObject(purchase);
+						// Return the object
+						mMakePurchaseCbContext.success(purchaseResult);
+					} catch (JSONException e) { }
+					// Clear the handler
+					mMakePurchaseCbContext = null;
+				}
+
+				// Set the purchase has pending in SharedPreferences
+				SharedPreferences pendingPurchasesCache = cordova.getActivity().getSharedPreferences(PENDING_PURCHASES_CACHE_NAME, 0);
+				SharedPreferences.Editor editor = pendingPurchasesCache.edit();
+				editor.putBoolean(purchase.getSku(), true);
+				editor.commit();
+
+				// Add the purchase to the inventory
+				mInventory.addPurchase(purchase);
 			}
-
-			// Set the purchase has pending in SharedPreferences
-			SharedPreferences pendingPurchasesCache = cordova.getActivity().getSharedPreferences(PENDING_PURCHASES_CACHE_NAME, 0);
-			SharedPreferences.Editor editor = pendingPurchasesCache.edit();
-			editor.putBoolean(purchase.getSku(), true);
-			editor.commit();
-
-			// Add the purchase to the inventory
-			mInventory.addPurchase(purchase);
 		}
 	};
 

